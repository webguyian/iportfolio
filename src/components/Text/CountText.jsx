import React from 'react';
import PropTypes from 'prop-types';

import Text from 'components/Text/Text';

const CountText = props => {
  const { children, count, emptyLabel, ...otherProps } = props;

  if (count === 1) {
    const label = children.slice(0, children.length - 1);

    return <Text {...otherProps}>{`${count} ${label}`}</Text>;
  }

  return <Text {...otherProps}>{`${count || emptyLabel} ${children}`}</Text>;
};

CountText.propTypes = {
  children: PropTypes.node.isRequired,
  count: PropTypes.number,
  emptyLabel: PropTypes.string
};

CountText.defaultProps = {
  count: 0,
  emptyLabel: 'No'
};

export default CountText;
