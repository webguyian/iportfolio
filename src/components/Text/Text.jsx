import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Text = props => {
  const { element, modifier, type, ...otherProps } = props;
  const baseClass = `ui-text ui-text--${type}`;
  const modifierClass = modifier && `ui-text--${modifier}`;

  return React.createElement(element, {
    className: classNames(baseClass, modifierClass),
    ...otherProps
  });
};

Text.propTypes = {
  element: PropTypes.string,
  modifier: PropTypes.oneOf(['block', 'bold']),
  type: PropTypes.oneOf(['accessible', 'body', 'display'])
};

Text.defaultProps = {
  element: 'span',
  type: 'body'
};

export default Text;
