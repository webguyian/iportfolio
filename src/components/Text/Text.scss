.ui-text {
  &--accessible {
    @include accessibly-hidden;
  }

  &--body {
    @include type-body;
  }

  &--display {
    @include type-display;
    display: block;
  }

  &--block {
    display: block;
    width: 100%;
  }

  &--bold {
    font-weight: $font-weight-bold;
  }

  &--light {
    font-weight: $font-weight-light;
  }

  &--loading {
    font-size: $font-size-l;
    text-align: center;
    animation: fade-in-out 2s linear infinite;
  }
}

@keyframes fade-in-out {
  0% {
    opacity: 1;
  }

  100% {
    opacity: 0.2;
  }
}
