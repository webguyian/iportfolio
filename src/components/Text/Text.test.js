import React from 'react';
import { create } from 'react-test-renderer';

import Text from './Text';

describe('<Text />', () => {
  it('renders correctly', () => {
    const component = create(<Text>Text goes here</Text>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with element prop', () => {
    const component = create(
      <Text element="h2" type="display">
        Heading
      </Text>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with modifier prop', () => {
    const component = create(<Text modifier="bold">Text goes here</Text>);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
