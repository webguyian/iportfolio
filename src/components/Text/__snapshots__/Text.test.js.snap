// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> renders correctly 1`] = `
<span
  className="ui-text ui-text--body"
>
  Text goes here
</span>
`;

exports[`<Text /> renders correctly with element prop 1`] = `
<h2
  className="ui-text ui-text--display"
>
  Heading
</h2>
`;

exports[`<Text /> renders correctly with modifier prop 1`] = `
<span
  className="ui-text ui-text--body ui-text--bold"
>
  Text goes here
</span>
`;
